solcjs

```diff
@@ -14,18 +14,37 @@ var smtsolver = require('./smtsolver.js');
 process.removeAllListeners('uncaughtException');
 var commander = require('commander');
 
+const parseOptimizeRuns = function(value) {
+  const parsedValue = parseInt(value, 10);
+  if (isNaN(parsedValue)) {
+    throw new Error('The "runs" setting must be an unsigned number');
+  }
+  return parsedValue;
+};
+
 var program = new commander.Command();
 program.name('solcjs');
 program.version(solc.version());
 program
   .option('--version', 'Show version and exit.')
   .option('--optimize', 'Enable bytecode optimizer.', false)
+  .option(
+    '--optimize-runs <optimize-runs>',
+    'The number of runs specifies roughly how often each opcode of the deployed code will be executed across the lifetime of the contract. ' +
+    'Lower values will optimize more for initial deployment cost, higher values will optimize more for high-frequency usage.',
+    parseOptimizeRuns
+  )
   .option('--bin', 'Binary of the contracts in hex.')
   .option('--abi', 'ABI of the contracts.')
   .option('--standard-json', 'Turn on Standard JSON Input / Output mode.')
   .option('--base-path <path>', 'Automatically resolve all imports inside the given path.')
   .option('-o, --output-dir <output-directory>', 'Output directory for the contracts.');
-program.parse(process.argv);
+
+try {
+  program.parse(process.argv);
+} catch(e) {
+  abort(e.message);
+}
```

Comment:
This should only catch validation errors. Unexpected exceptions should cause a crash.

Reply:
Thanks, have updated accordingly.

Reply:
@rorye Can you rebase on `master` and squash review fixes?

Reply:
@cameel please see update.